### PR TITLE
Export CRL and Database CA in DER format

### PR DESCRIPTION
**Pull request by gabrielcorado**
Related to #13837.

Adds a new format to `tctl auth export` to export the database CA to DER format. This will be imported into Microsoft AD (similar to what is done for desktop access).

It also adds a new command to export certificate revocation lists (CRL) `tctl auth crl`. Since we won't initially support updating the CRLs on the Teleport server ([like it is done for desktop access](https://github.com/gravitational/teleport/blob/master/lib/auth/windows/certificate_authority.go#L159)), it must be published on the Windows certificate store.

### CODE REVIEW

File: tool/tctl/common/auth_command.go [hunk 1]
```diff
@@ -144,6 +146,9 @@ func (a *AuthCommand) Initialize(app *kingpin.Application, config *service.Confi
 
 	a.authLS = auth.Command("ls", "List connected auth servers")
 	a.authLS.Flag("format", "Output format: 'yaml', 'json' or 'text'").Default(teleport.YAML).StringVar(&a.format)
+
+	a.authCRL = auth.Command("crl", "Export certificate revocation list (CRL) for certificates authorities")
```
Comment (zmb3) [inline tool/tctl/common/auth_command.go:150]:
```suggestion
	a.authCRL = auth.Command("crl", "Export certificate revocation list (CRL) for certificate authorities")
```
We always generate an empty CRL, right? Should the help message indicate such?

File: tool/tctl/common/auth_command.go [hunk 2]
```diff
@@ -394,6 +410,23 @@ func (a *AuthCommand) ListAuthServers(ctx context.Context, clusterAPI auth.Clien
...
+	if err := certType.Check(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	crl, err := clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	fmt.Println(string(crl))
```
Comment (zmb3) [inline tool/tctl/common/auth_command.go:426]: Does this write binary output to the terminal?

Comment (gabrielcorado) [reply tool/tctl/common/auth_command.go:426]: Yes. It is the same as how we're printing the contents on [`auth export`](https://github.com/gravitational/teleport/blob/master/lib/client/ca_export.go#L268).

Also, I've verified the contents with `openssl crl -in db.crl -inform DER -text -noout`:
```
Certificate Revocation List (CRL):
        Version 2 (0x1)
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: /O=root.teleport.dev/CN=root.teleport.dev/serialNumber=000000000000000000000000000000000000000
        Last Update: Mar  3 00:00:00 2023 GMT
        Next Update: Mar  2 00:00:00 2024 GMT
...
```

File: tool/tctl/common/auth_command.go (hunk 1 above)
Comment (gabrielcorado) [reply tool/tctl/common/auth_command.go:150]: > We always generate an empty CRL, right? Should the help message indicate such?

Right, I'll update the message to indicate that.

File: tool/tctl/common/auth_command.go (hunk 2 above)
Comment (zmb3) [reply tool/tctl/common/auth_command.go:426]: Couldn't that have side effects, like including control characters that change the color of the terminal or put it in raw mode?

Comment (gabrielcorado) [reply tool/tctl/common/auth_command.go:426]: I could. As I can see, we're usually using the command on our docs to direct the output to a file (e.g.: [`tctl auth export --type=windows > user-ca.der`](https://goteleport.com/docs/desktop-access/active-directory-manual/#export-the-teleport-ca)). Do you think it is better to have a `--out/-o` as we do on `tctl auth sign`? In that case, users could still output to stdout and direct it to a file.

File: tool/tctl/common/auth_command.go [hunk 3]
```diff
@@ -394,6 +410,23 @@ func (a *AuthCommand) ListAuthServers(ctx context.Context, clusterAPI auth.Clien
 	return nil
 }
 
+// GenerateCRLForCA generates a certificate revocation list for a certificate
+// authoritity.
```
Comment (smallinsky) [inline tool/tctl/common/auth_command.go:414]:
```suggestion
// authority.
```

File: tool/tctl/common/auth_command.go [hunk 4]
```diff
@@ -394,6 +410,23 @@ func (a *AuthCommand) ListAuthServers(ctx context.Context, clusterAPI auth.Clien
...
+func (a *AuthCommand) GenerateCRLForCA(ctx context.Context, clusterAPI auth.ClientI) error {
+	certType := types.CertAuthType(a.caType)
+	if err := certType.Check(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	crl, err := clusterAPI.GenerateCertAuthorityCRL(ctx, certType)
```
Comment (smallinsky) [inline tool/tctl/common/auth_command.go:421]: Don't we need to align the RBAC role check ? 
https://github.com/gravitational/teleport/blob/6973cd652720011c2fb6466cfe0889533d9dfae8/lib/auth/auth_with_roles.go#L5549

**Review by zmb3 — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]

**Pull request merged**